rmc-docs/src/limitations.md

```diff
@@ -0,0 +1,413 @@
+10.1.1 | Boolean type | Yes | |
+10.1.2 | Numeric types | Yes | |
+10.1.3 | Textual types | Yes | |
+10.1.4 | Never type | Yes | |
+10.1.5 | Tuple types | Yes | |
+10.1.6 | Array types | Yes | |
+10.1.7 | Slice types | Yes | |
+10.1.8 | Struct types | Yes | |
+10.1.9 | Enumerated types | Yes | |
+10.1.10 | Union types | Yes | |
+10.1.11 | Function item types | Yes | |
+10.1.12 | Closure types | Partial | See [Observations - Advanced features](#advanced-features) |
+10.1.13 | Pointer types | Partial | See [Observations - Advanced features](#advanced-features) |
+10.1.14 | Function pointer types | Partial | See [Observations - Advanced features](#advanced-features) |
+10.1.15 | Trait object types | Partial | See [Observations - Advanced features](#advanced-features) |
+10.1.16 | Impl trait type | Partial | See [Observations - Advanced features](#advanced-features) |
+10.1.17 | Type parameters | Partial | See [Observations - Advanced features](#advanced-features) |
+10.1.18 | Inferred type | Partial | See [Observations - Advanced features](#advanced-features) |
+10.2 | Dynamically Sized Types | Partial | See [Observations - Advanced features](#advanced-features) |
+10.3 | Type layout | Yes | |
+10.4 | Interior mutability | Yes | |
+10.5 | Subtyping and Variance | Yes | |
+10.6 | Trait and lifetime bounds | Yes | |
+10.7 | Type coercions | Partial | See [Observations - Advanced features](#advanced-features) |
+10.8 | Destructors | Partial | |
+10.9 | Lifetime elision | Yes | |
+11 | Special types and traits | Partial | |
+| | `Box<T>` | Yes | |
+| | `Rc<T>` | Yes | |
+| | `Arc<T>` | Yes | |
+| | `Pin<T>` | Yes | |
+| | `UnsafeCell<T>` | Partial | |
+| | `PhantomData<T>` | Partial | *Review this* |
+| | Operator Traits | Partial | |
+| | `Deref` and `DerefMut` | Yes | |
+| | `Drop` | Partial | |
+| | `Copy` | Yes | |
+| | `Clone` | Yes | |
+14 | Linkage | Yes | |
+15.1 | Unsafe functions | Yes | |
+15.2 | Unsafe blocks | Yes | |
+15.3 | Behavior considered undefined | Partial | |
+| | Data races | No | See [Observations - Concurrency](#concurrency) |
+| | Dereferencing dangling raw pointers | Yes | |
+| | Dereferencing unaligned raw pointers | No | |
+| | Breaking pointer aliasing rules | No | |
+| | Mutating immutable data | No | |
+| | Invoking undefined behavior via compiler intrinsics | Partial | See [Observations - Intrinsics](#intrinsics) |
+| | Executing code compiled with platform features that the current platform does not support | No | |
+| | Producing an invalid value, even in private fields and locals | No | |
+
+## Observations on partially or unsupported features
+
+### Code generation for unsupported features
+
+RMC aims to be a industrial verification tool. Most industrial crates may
+include unsupported features in parts of their code that do not need to be
+verified. In general, this should not prevent users from verifying their code.
+
+Because of that, the general rule is that RMC generates an `assert(false)`
```

Comment:
Technically, it generates `assert(false)` followed by `assume(false)`, I think this piece of detail is worth mentioning since it may cause some asserts to become unreachable, and hence vacuously pass.